### ISSUE TITLE
array event index not updated

### ISSUE THREAD

**Issue by sabob** [ractivejs/ractive#2399]
This fiddle: http://jsfiddle.net/sabob/m8325w9a/2/ shows that the event's index references passed from the template is not updated after array items are removed.

In the fiddle, click on the Delete button next to the top item, in the console note the index is 0.
Click Delete again of the new top item, in the console note the index is 1, instead of 0. Also, instead of deleting "John", "Steve" was deleted.

Works in 0.7.3

**Comment by fskreuz** [comment]
BUG! A workaround for now would be to get the index from the keypath `e.keypath.split('.').pop()`.

http://jsfiddle.net/m8325w9a/4/

```js
var view = new Ractive({
	el: 'body',
  template: `
  	<ul>
      {{#each items:i}}
      <li>
        {{name}}
        <button on-click="remove(event, this)">Delete</button>
      </li>
      {{/each}}
    </ul>
  `,
  data: {
    items: [
      {name: "Bob"},
      {name: "John"},
      {name: "Steve"}
    ]
  },
  remove: function(e, obj) {
    console.log("index: %s keypath: %s", e.index.i, e.keypath);
    this.splice("items", e.index.i, 1);
	}
})
```

**Comment by woodlandhunter** [comment]
Another workaround could be to pass the index directly: `on-click="remove(event, this, i)`

**Comment by sabob** [comment]
Thanks, I'm actually trying to get this decorator to work in edge as I use it in 0.7.3:
https://github.com/ractivejs/Ractive-decorators-sortable

The event index not updating is one bug, but it seems there are more issues. I'll update as I go along.

**Comment by sabob** [comment]
Can we tag this issue under milestone 0.0.8?

**Comment by sabob** [comment]
fixed by #2402
